src/main/java/software/amazon/cloudformation/resource/TypeConverter.java

```diff
@@ -0,0 +1,49 @@
+/*
+* Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+package software.amazon.cloudformation.resource;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.io.IOException;
+
+public class TypeConverter {
+    protected TypeConverter() {
+        throw new UnsupportedOperationException();
+    }
+
+    public static Object covertProperty(final Object input, final TypeReference<?>... typeReferences) // preserves order
+    {
+        try {
+            final Serializer serializer = new Serializer();
+            final String stringified_input = serializer.serialize(input);
+
+            int iter = 0;
+            while (true) {
+                TypeReference<?> typeReference = typeReferences[iter];
+                try {
+                    return serializer.deserialize(stringified_input, typeReference);
```

Comment:
what happens if a multi-typed property has types string and int? would the ordering of how we specify the potential references matter? (not sure if this is the case) if a type correctly deserializes under both typeReferences passed into this function, why take the first over the second?

Reply:
yep the idea is to preserve the order